lib/view_component/collection.rb

```diff
@@ -26,5 +28,25 @@ def collection_variable(object)
         raise ArgumentError.new("The value of the argument isn't a valid collection. Make sure it responds to to_ary: #{object.inspect}")
       end
     end
+
+    def collection_parameter_name
+      @component.collection_parameter_name
+    end
+
+    def collection_counter_parameter_name
+      "#{collection_parameter_name}_counter".to_sym
+    end
+
+    def component_options(item, iterator)
+      @options[collection_parameter_name]         = item
+      @options[collection_counter_parameter_name] = iterator.index if counter_argument_present?
+      # @options.merge!(:"#{collection_parameter_name}" => item)
+      # @options.merge!(:"#{collection_counter_parameter_name}" => iterator.index) if counter_argument_present?
+      @options
+    end
```

Comment:
I don't think we want to change `@options` here. Lets leave it as set.

```suggestion
  def component_options(item, iterator)
    item_options = { collection_parameter_name => item }
    item_options[collection_counter_parameter_name] = iterator.index if counter_argument_present?
    @options.merge(item_options)
  end
```